internal/kibana/ingestmanager/client_packages.go

```diff
@@ -0,0 +1,46 @@
+// Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+// or more contributor license agreements. Licensed under the Elastic License;
+// you may not use this file except in compliance with the Elastic License.
+
+package ingestmanager
+
+import (
+	"encoding/json"
+	"fmt"
+
+	"github.com/pkg/errors"
+
+	"github.com/elastic/elastic-package/internal/packages"
+)
+
+// InstallPackage installs the given package in Fleet.
+func (c *Client) InstallPackage(pkg packages.PackageManifest) ([]packages.Asset, error) {
+	return managePackage(pkg, "install", c.post)
+}
+
+// RemovePackage removes the given package from Fleet.
+func (c *Client) RemovePackage(pkg packages.PackageManifest) ([]packages.Asset, error) {
+	return managePackage(pkg, "remove", c.delete)
```

Comment:
I have a gut feeling that clients in `ingestmanager` are too much complex. Not sure if we need as many layers to process a single request. Maybe not depending on common abstraction would be simpler?

Please take a look at https://github.com/elastic/elastic-package/blob/master/internal/kibana/saved_objects.go . In this case I need to implement paging and the exact code responsible for calls to Kibana is wrapped up in ca. 20 LOCs: https://github.com/elastic/elastic-package/blob/master/internal/kibana/saved_objects.go#L95-L118

I'm happy to discuss it!